src/components/StoryBook/FollowButton.tsx

```diff
@@ -0,0 +1,88 @@
+import { Button, ButtonProps, styled } from '@mui/material';
+import { useEffect, useState } from 'react';
+import { useNavigate, useParams } from 'react-router-dom';
+
+import { createFollow, removeFollow } from '../../apis/follow';
+import { userInfo } from '../../apis/userInfo';
+import { USER_ID_KEY } from '../../constants/auth';
+import { ROUTES } from '../../constants/routes';
+import { Follow as Following } from '../../interfaces/user';
+import { getLocalStorage } from '../../utils/storage';
+
+const FollowButton = () => {
+  const [isFollowing, setIsFollowing] = useState(false);
+  const [canBeRendered, setCanBeRendered] = useState(false);
+  const [followId, setFollowId] = useState('');
+  const { userId } = useParams();
+  const navigate = useNavigate();
+
+  const storedUserId = getLocalStorage(USER_ID_KEY);
+  const isMyStoryBook = userId === storedUserId;
+
+  const handleClick = async () => {
+    if (isFollowing && userId && followId) {
+      const result = await removeFollow(followId);
+      result && setIsFollowing(false);
+      return;
+    }
+
+    if (userId) {
+      const result = await createFollow(userId);
+      if (result) {
+        const {
+          data: { _id: newFollowId },
+        }: { data: { _id: string } } = result;
+        newFollowId && setFollowId(newFollowId);
+        newFollowId && setIsFollowing(true);
```

Comment:
```suggestion
 if (newFollowId) {
   setFollowId(newFollowId);
   setIsFollowing(true);
 }
```
코드 줄이 길어지지만 이런식으로 사용할 수도있을것 같아요